core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileLogger.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.utils.files
+
+import android.content.Context
+import android.util.Log
+import java.io.File
+import java.io.IOException
+import java.lang.System.currentTimeMillis
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Authored by s-ayush2903 on 19 June 2020 */
+
+/** A class for writing logs to the file in the device */
+@Singleton
+class FileLogger @Inject constructor() {
+
+  fun writeLogFile(context: Context): File {
+    // Create a new folder in private storage with name: logs
+    val logDir = File(context.filesDir, "logs")
+    val logFile = File(logDir, fileName)
+
+    Log.d("KIWIX", "Writing all logs into [" + logDir.path + "]")
+
+    if (!logDir.exists()) {
+      logDir.mkdir()
+    }
+    // Delete the pre-existing file
+    if (logDir.exists() && logFile.isFile) {
+      Log.d(TAG, "writeLogFile: Deleting preExistingFile")
+      logFile.delete()
+    }
+
+    try {
+      logDir.createNewFile()
+      Runtime.getRuntime().exec("logcat -f $logFile")
+      Runtime.getRuntime().exec("logcat -b all -d")
+    } catch (e: IOException) {
+      Log.e("KIWIX", "Error while writing $fileName! ", e)
+    }
+    return logFile
+  }
+
+  companion object {
+    private const val TAG = "FileLogger"
+    private val fileName: String
+      get() = "logs" + currentTimeMillis() + ".txt"
```

Comment:
I feel that these (static)constants can contribute to MemoryLeak. So this should better be changed

Reply:
this is not static. It cannot be accessed statically even if it wasn't private, eg `FileLogger.filename`, for a memory leak to occur we would need to hold a reference to an object longer than the lifetime of said object, there is an increased risk of that happening with static references for sure but no object is referenced here.

Static functionality is "achieved" with the `object` keyword in kotlin, to add some "statics" to a class you would use a `companion object`, I believe there was an android developers "kotlin vocabulary" video released recently enough on the `object` topic, maybe give it a [watch](https://www.youtube.com/watch?v=KUk6k865Vgg) also the [docs](https://kotlinlang.org/docs/reference/object-declarations.html)

Reply:
ah my bad, the preview on the conversation tab didn't show the companion object declaration above.

However there is no memory leak.